### DISCUSSION TITLE
Teaching "boring" syntax

### DISCUSSION THREAD

**Discussion by Felienne** [hedyorg/hedy#785] [Language Ideas]
@MarleenGilsing and I just had a wonderful conversation about teaching syntax. We are struggling with the issue that some of the higher levels are pretty boring and only teach syntax. That violates design goal DG6, because the changes in syntax then do not really add new meaningful programs.

An example of a *meaningful* addition is the current level 3. Adding quotation marks in level 3 enables new types of programs. An example of a less meaningful addition is the current level 9, which adds the colon. The colon does not make programs more fun or even allow for new types of programs.

How to approach this? We have a few directions in mind:
* Following design goal DG4, we could move syntactic elements even later, potentially (idea from Marleen) combine a few into one level under the guise of "now we are going to learn real real Python"
* Alternatively, we could combine commands and syntax, for example combining `elif` with the colon (for `elif` and for `if`) That has as a downside that we might detract attention from students, but does help with making cool programs.
* Alternatively alternatively, we could step away from programming as an activity to teach syntax... Maybe a little quiz is a better [this idea is going to go in a separate discussion too]. In that case we could mix syntax-based quiz levels training syntax with free programming levels exploring concepts.

Relevant design goals for this discussion are below. Or consult the [full design](https://github.com/Felienne/hedy/blob/main/DESIGN.md).

- DG4 Adding syntactic elements like brackets and colons is deferred to the latest moment possible
- DG6 At every level it is possible to create simple but meaningful programs

**Comment by das-g** [comment]
Would it be an option to modify [design goal](https://github.com/Felienne/hedy/blob/main/DESIGN.md#design-goals) 4 and exclude colons from it, i.e. allow them earlier than at "the latest moment possible"?

While in natural language, colons might be a punctuation mark taught, learned and acquired somewhat late, my impression is (Please correct me, if I'm wrong.) that Hedy assumes that learners are already sufficiently progressed in natural language writing to use colons naturally and intuitively where they belong.

So when introducing syntax, we can maybe (where appropriate) build upon learners' knowledge of natural language grammar.

Level-7 code like
```
repeat 3 times print 'Hedy is fun!'
```
feels kinda awkward to me. Not because of its difference to Python or other programming languages, but because of its difference to (written) natural language: I'd really expect _some_ punctuation mark between "times" and "print". (Well, not just _any_ punctuation mark, but either a comma or a colon.)

```
repeat 3 times: print 'Hedy is fun!'
```
feels much closer to how I'd express that in natural English.

So maybe this (and other) control structures can be introduced not just with their keyword(s), but also with (a bit) more syntax, including (at least some of) the colons.

**Reply by das-g** [reply]
(For `if` this might be trickier, as conditional sentences in English don't always require internal punctuation and because the usual punctuation there (if any) would almost always be a comma, _not_ a colon.)
